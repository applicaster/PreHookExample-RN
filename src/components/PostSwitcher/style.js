import { StyleSheet } from 'react-native';

const optionSize = 30;
const iconSize = 18;

export const styles = StyleSheet.create({
  postSwitcherContainer: {
    flexDirection: 'row',
    borderRadius: 5,
    borderWidth: 1,
    marginRight: 11,
    height: optionSize,
    width: optionSize * 2,
  },
  facebookOption: {
    flex: 0.5,
    alignItems: 'center',
    justifyContent: 'center',
  },
  twitterOption: {
    flex: 0.5,
    alignItems: 'center',
    justifyContent: 'center',
  },
  facebookOptionImage: {
    height: iconSize,
    width: iconSize,
  },
  twitterOptionImage: {
    height: iconSize,
    width: iconSize,
  },
});
