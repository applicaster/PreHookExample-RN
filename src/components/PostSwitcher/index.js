import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Image,
  View,
  TouchableWithoutFeedback,
} from 'react-native';
import { styles } from './style';

const FB_POST_ICON = 'feed_rn_fb_post_icon';
const TWITTER_POST_ICON = 'feed_rn_twitter_post_icon';

class PostSwitcher extends Component {
  render() {
    const { selectedOption } = this.props;
    const { backgroundColor, mainColor, textColor } = this.context;

    const selectedTintColor = { tintColor: textColor };
    const unSelectedTintColor = { tintColor: textColor };

    const selectedBackgroundColor = { backgroundColor: mainColor };
    const unSelectedBackgroundColor = { backgroundColor };

    const borderColorStyle = { borderColor: mainColor };

    return (
      <View style={[styles.postSwitcherContainer, borderColorStyle]}>
        <TouchableWithoutFeedback style={[styles.facebookOption]}>
          <Image style={styles.facebookOptionImage} source={{ uri: FB_POST_ICON }} />
        </TouchableWithoutFeedback>
        <TouchableWithoutFeedback style={[styles.twitterOption]}>
          <Image style={styles.twitterOptionImage} source={{ uri: TWITTER_POST_ICON }} />
        </TouchableWithoutFeedback>
      </View>
    );
  }
}

PostSwitcher.propTypes = {
  selectedOption: PropTypes.string,
};

PostSwitcher.contextTypes = {
  backgroundColor: PropTypes.string,
  mainColor: PropTypes.string,
  textColor: PropTypes.string,
};

export default PostSwitcher;
