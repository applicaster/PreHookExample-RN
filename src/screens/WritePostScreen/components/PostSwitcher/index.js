import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Image,
  View,
  TouchableWithoutFeedback,
} from 'react-native';
import { styles } from './style';

const FB_POST_ICON = 'feed_rn_fb_post_icon';
const TWITTER_POST_ICON = 'feed_rn_twitter_post_icon';

class PostSwitcher extends Component {
  render() {
    const { socialNetworkSelected, toggleNetworkSelected } = this.props;
    const { mainColor, textColor } = this.context.styles;
    const borderColorStyle = { borderColor: mainColor };
    const tintColorStyle = { tintColor: textColor };
    const twitterSelectedStyles = { backgroundColor: (socialNetworkSelected === 'twitter') ? mainColor : 'transparent' };
    const facebookSelectedStyles = { backgroundColor: (socialNetworkSelected === 'facebook') ? mainColor : 'transparent' };

    return (
      <View style={[styles.postSwitcherContainer, borderColorStyle]}>
        <TouchableWithoutFeedback onPress={toggleNetworkSelected}>
          <View style={[styles.facebookOption, facebookSelectedStyles]}>
            <Image style={[styles.facebookOptionImage, tintColorStyle]} source={{ uri: FB_POST_ICON }} />
          </View>
        </TouchableWithoutFeedback>
        <TouchableWithoutFeedback onPress={toggleNetworkSelected}>
          <View style={[styles.twitterOption, twitterSelectedStyles]}>
            <Image style={[styles.twitterOptionImage, tintColorStyle]} source={{ uri: TWITTER_POST_ICON }} />
          </View>
        </TouchableWithoutFeedback>
      </View>
    );
  }
}

PostSwitcher.propTypes = {
  socialNetworkSelected: PropTypes.string,
  toggleNetworkSelected: PropTypes.func,
};

PostSwitcher.contextTypes = {
  styles: PropTypes.object,
};

export default PostSwitcher;
