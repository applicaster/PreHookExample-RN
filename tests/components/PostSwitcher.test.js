import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import PostSwitcher from '../../src/components/PostSwitcher';

describe('PostSwitcher', () => {
  test('component renders correctly when facebook is selected', () => {
    const html = renderer.create(
      <PostSwitcher socialNetworkSelected={'facebook'} toggleNetworkSelected={() => {}} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly when twitter is selected', () => {
    const html = renderer.create(
      <PostSwitcher socialNetworkSelected={'twitter'} toggleNetworkSelected={() => {}} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });
});
