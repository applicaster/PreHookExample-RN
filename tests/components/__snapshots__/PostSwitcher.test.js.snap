// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostSwitcher component renders correctly when facebook is selected 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 30,
        "marginRight": 11,
        "width": 60,
      },
      Object {
        "borderColor": undefined,
      },
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 0.5,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": undefined,
        },
      ]
    }
    testID={undefined}
  >
    <Image
      source={
        Object {
          "uri": "feed_rn_fb_post_icon",
        }
      }
      style={
        Array [
          Object {
            "height": 18,
            "width": 18,
          },
          Object {
            "tintColor": undefined,
          },
        ]
      }
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 0.5,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "transparent",
        },
      ]
    }
    testID={undefined}
  >
    <Image
      source={
        Object {
          "uri": "feed_rn_twitter_post_icon",
        }
      }
      style={
        Array [
          Object {
            "height": 18,
            "width": 18,
          },
          Object {
            "tintColor": undefined,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`PostSwitcher component renders correctly when twitter is selected 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 30,
        "marginRight": 11,
        "width": 60,
      },
      Object {
        "borderColor": undefined,
      },
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 0.5,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "transparent",
        },
      ]
    }
    testID={undefined}
  >
    <Image
      source={
        Object {
          "uri": "feed_rn_fb_post_icon",
        }
      }
      style={
        Array [
          Object {
            "height": 18,
            "width": 18,
          },
          Object {
            "tintColor": undefined,
          },
        ]
      }
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 0.5,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": undefined,
        },
      ]
    }
    testID={undefined}
  >
    <Image
      source={
        Object {
          "uri": "feed_rn_twitter_post_icon",
        }
      }
      style={
        Array [
          Object {
            "height": 18,
            "width": 18,
          },
          Object {
            "tintColor": undefined,
          },
        ]
      }
    />
  </View>
</View>
`;
